File: trunk/qooxdoo-contrib/QWT/trunk/sushi/core/src/main/java/org/qooxdoo/sushi/metadata/store/Store.java
/* ************************************************************************
   
   qooxdoo - the new era of web development
   
   http://qooxdoo.org
   
   Copyright:
     2006-2008 1&1 Internet AG, Germany, http://www.1and1.org
   
   License:
     LGPL: http://www.gnu.org/licenses/lgpl.html
     EPL: http://www.eclipse.org/org/documents/epl-v10.php
     See the LICENSE file in the project's top-level directory for details.
   
   Authors:
     * Michael Hartmeier (mlhartme)
   
 ************************************************************************ */

package org.qooxdoo.sushi.metadata.store;

import java.util.List;

import org.qooxdoo.sushi.metadata.Item;

public interface Store {
    void write(List<Item<?>> parents, String path, String value) throws Exception;
    
    /** @return null if not found */ 
    String read(List<Item<?>> parents, String path) throws Exception;
}

File: trunk/qooxdoo-contrib/Persist/trunk/source/class/persist/Store.js
/* ************************************************************************

   Copyright:
     2007-2010 Christian Boulanger

   License:
     LGPL: http://www.gnu.org/licenses/lgpl.html
     EPL: http://www.eclipse.org/org/documents/epl-v10.php
     See the LICENSE file in the project's top-level directory for details.

   Authors:
     * Christian Boulanger (cboulanger)

************************************************************************ */

/* ************************************************************************
#ignore(Persist.Store)
************************************************************************ */
/**
 * This is a thin wrapper around the persistjs library
 * See http://pablotron.org/software/persist-js/
 */
qx.Class.define("persist.Store",
{
  extend : qx.core.Object,
  
  /*
  *****************************************************************************
     CONSTRUCTOR
  *****************************************************************************
  */

  /**
   * Contructor. Creates a new browser store
   * @param name {String}
   */
  construct : function( name )
  {
    this.base(arguments);
    this.__store = new Persist.Store(name);
  },


  /*
  *****************************************************************************
     MEMBERS
  *****************************************************************************
  */  
  members :
  {
    
    /*
    ---------------------------------------------------------------------------
       PRIVATE MEMBERS
    ---------------------------------------------------------------------------
    */       
    __store : null,
    
    /*
    ---------------------------------------------------------------------------
       API METHODS
    ---------------------------------------------------------------------------
    */
    
    /**
     * Fetches the value stored under the given key and calls the callback
     * function with the value. The callback is necessary for cross-browser
     * compatibility
     * 
     * @param key {String}
     * @param callback {Function}
     */
    get : function( key, callback )
    {
      this.__store.get( key, callback );
    },
    
    /**
     * Saves the given value to the store under the given key
     * @param key {String}
     * @param value {String}
     */
    set : function ( key, value )
    {
      this.__store.set( key, value );
    }
  }
});